### ISSUE TITLE
io.load_image_from_dir needs documentation/support

### ISSUE THREAD

**Issue by CameronTEllis** [brainiak/brainiak#354]
This is a critical function for brainiak but from the documentation or comments it is unclear how it works. Our code was not working because the files had been loaded in 'out of order' because this function was using the file order system of our OS (1, 10, 11, 12, ... 19, 20, 21, ...) rather than what we wanted (1, 2, 3 ...). It would be good if there were three things:

1. A way to print out the order of files that are loaded in (if it is currently possible, it could be easier or documented).
2. A comment in the docstring stating how the data is expected to be organized.
3. Encourage users to name their subjects with zeros (e.g. 01, 02, 03)

**Comment by mihaic** [comment]
The file names are sored lexicographically, so "10" comes before "2", as you noticed. This is how software I am familiar with works.

You can see the file names in the log as the loading happens if you set the logging level to debug.

I think extending the docstring the way you suggest would help. Looking forward to a PR.

**Comment by mihaic** [comment]
I think adding a sort option to enable "natural" sorting ("2" before "10") also makes sense. Nevertheless, I would leave lexicographic sorting as default.

**Comment by CameronTEllis** [comment]
I agree that the default should be the lexicographic ordering (especially since we don't want to change the behavior for other users). I think adding a natural sorting step would be a great idea, and more documentation to say that the information is available in the log file. Still though, why not make the listed paths an attribute of 'images'?

**Comment by mihaic** [comment]
The resulting images are Nibabel `SpatialImage`s (as mentioned in the documentation). You can get the filename of each image using `get_filename()`.
